packages/core/src/next-app-functions/generate-meta-data.ts

```diff
@@ -36,19 +39,20 @@ export async function generateMetadata({
 
   handleRequiredEnvs();
 
+  const params = await rawParams; // ✅ Await the params here
   const uri = params?.paths?.join("/") || "/";
   const siteSettings = await getSiteSettings();
   const { data, archive } = await getPageData(uri);
 
   const yoast =
-    data?.yoast_head_json || // seo data for single items (pages,posts,custom post type singles)
-    data?.page?.yoast_head_json || //  seo data for the page with matching slug to the archive (example, there is a page with slug "all-movies", and a movie post type with archive slug of "all-movies")
-    archive?.yoast_head_json; // seo data for the archive, if there is not a page with the same slug as the archive slug it will fallback to this
```

Comment:
any reason for removing those comments?

Reply:
i'll add them back in, bad habit of mine

Reply:
Probably good to keep them in this case because of the complicated resolution of the specific seo data / fallbacks.